vmware-tanzu/tanzu-framework
# Refactor package plugin to remove CreateSecret & CreateServiceAccount flag

**What this PR does / why we need it**:
The way we use CreateSecret flag is not appropriate in package install & update, remove this flag and use a boolean to control secret creation.

**Describe testing done for PR**:
Manually test on cluster

```
1. Package install testing
tanzu package install myfb --package-name fluent-bit.tanzu.vmware.com --version 1.7.5+vmware.1-tkg.1 -n test-ns
| Installing package 'fluent-bit.tanzu.vmware.com'
/ Getting package metadata for fluent-bit.tanzu.vmware.com
| Creating package resource
\ Package install status: Reconciling
 Added installed package 'myfb' in namespace 'test-ns'

2. Package install testing with creating new secret
tanzu package install myfb --package-name fluent-bit.tanzu.vmware.com -n test-ns --version 1.7.5+vmware.1-tkg.1 --values-file config.yaml
| Installing package 'fluent-bit.tanzu.vmware.com'
/ Getting package metadata for fluent-bit.tanzu.vmware.com
- Creating secret 'myfb-test-ns-values'
| Creating package resource
- Package install status: Reconciling

 Added installed package 'myfb' in namespace 'test-ns'

check secret myfb-test-ns-values content and matches config.yaml

3. Package update testing
tanzu package installed update myfb --version 1.7.5+vmware.1-tkg.1 -n test-ns --values-file newfile.yaml
| Updating package 'myfb'
- Getting package install for 'myfb'
\ Updating package install for 'myfb'
 Updated package install 'myfb' in namespace 'test-ns'

check secret myfb-test-ns-values content and matches newfile.yaml
```

**New PR Checklist**
